src/server/server.rs

```diff
@@ -728,4 +733,40 @@ mod tests {
         ch.try_send(Msg::Quit).unwrap();
         h.join().unwrap();
     }
+
+    #[test]
+    fn test_report_unreachable() {
+        let addr = "127.0.0.1:0".parse().unwrap();
+        let listener = TcpListener::bind(&addr).unwrap();
+        let resolver = MockResolver { addr: listener.local_addr().unwrap() };
+
+        let cfg = Config::new();
+
+        let evloop = store::create_event_loop::<ServerTransport, RpcClient>(&cfg.raft_store);
+        let sendch = SendCh::new(evloop.unwrap().channel());
+        let router = ServerRaftStoreRouter::new(sendch, 0);
+
+        let mut event_loop = create_event_loop(&cfg).unwrap();
+        let mut storage = Storage::new(&cfg.storage).unwrap();
+        storage.start(&cfg.storage).unwrap();
+
+        let server = Server::new(&mut event_loop,
+                                 &cfg,
+                                 listener,
+                                 storage,
+                                 router,
+                                 resolver,
+                                 store::new_snap_mgr("", None))
+            .unwrap();
+
+        for i in 0..10 {
+            let mut msg = Message::new();
+            if i % 2 == 1 {
+                msg.set_raft(RaftMessage::new());
+            }
+            server.report_unreachable(ConnData::new(0, msg));
+            let counter = RAFT_STORE_MSG_COUNTER.with_label_values(&["unreachable"]).get();
```

Comment:
RAFT_STORE_MSG_COUNTER is a global value, other tests may also add it, so we can't use this for check directly.